spring-cloud-stream/src/main/java/org/springframework/cloud/stream/annotation/EnableModule.java

```diff
@@ -47,7 +46,7 @@
 @Configuration
 @MessageEndpoint
 @Import({RedisServiceConfiguration.class, RabbitServiceConfiguration.class,
-	ChannelBindingAdapterConfiguration.class, CodecConfiguration.class, ChannelBindingAdapterRunner.class,
```

Comment:
spring.factories would also work for CodecConfiguration. Currently this class is explicitly imported in the Binder configurations as described in the PR description. An alternate implementation would have to provide a CodecConfiguration.  spring.factories is more flexible but it would be harder to find where the Codec is configured.